src/common/reconciler.go

```diff
@@ -58,234 +77,175 @@ func (r *ServiceReconciler) Reconcile(registration opslevel_jq_parser.ServiceReg
-			if expected.Key == match.Key && expected.Value == match.Value {
-				found[i] = true
-				break
-			}
-		}
+	if len(tagAssigns) > len(serviceTags) {
+		return false
+	}
+	serviceTagsMap := make(map[string]bool)
+	for _, tag := range serviceTags {
+		serviceTagsMap[tag.Key+tag.Value] = true
 	}
-	for _, value := range found {
-		if !value {
+	for _, tag := range tagAssigns {
+		if _, ok := serviceTagsMap[tag.Key+tag.Value]; !ok {
 			return false
 		}
 	}
 	return true
 }
 
 func (r *ServiceReconciler) ServiceNeedsUpdate(input opslevel.ServiceUpdateInput, service *opslevel.Service) bool {
-	if input.Name != nil && *input.Name != service.Name {
+	// TODO: we don't support the system/parent on service??????????
+	if input.Description != nil && *input.Description != service.Description {
 		return true
 	}
-	if input.Product != nil && *input.Product != service.Product {
+	if input.Framework != nil && *input.Framework != service.Framework {
 		return true
 	}
-	if input.Description != nil && *input.Description != service.Description {
+	if input.Language != nil && *input.Language != service.Language {
 		return true
 	}
-	if input.Language != nil && *input.Language != service.Language {
+	if input.LifecycleAlias != nil && *input.LifecycleAlias != service.Lifecycle.Alias {
 		return true
 	}
-	if input.Framework != nil && *input.Framework != service.Framework {
+	if input.Name != nil && *input.Name != service.Name {
 		return true
 	}
-	if input.TierAlias != nil && *input.TierAlias != service.Tier.Alias {
+	if input.OwnerInput != nil && *input.OwnerInput.Alias != service.Owner.Alias {
 		return true
 	}
-	if input.LifecycleAlias != nil && *input.LifecycleAlias != service.Lifecycle.Alias {
+	if input.Product != nil && *input.Product != service.Product {
 		return true
 	}
-	if input.OwnerInput != nil && *input.OwnerInput.Alias != service.Owner.Alias {
+	if input.TierAlias != nil && *input.TierAlias != service.Tier.Alias {
 		return true
 	}
 	return false
 }
 
-// This function has 4 outcomes that can happen while looping over the aliases list
+// lookupService has 4 outcomes that can happen while looping over the aliases list
 // serviceAliasesResult_NoAliasesMatched - means that all API calls succeeded and none of the aliases matched an existing service
 // serviceAliasesResult_AliasMatched - means that all the API calls succeeded and a single service was found matching 1 of N aliases
 // serviceAliasesResult_MultipleServicesFound - means that all API calls succeeded but multiple services were returning means the list of aliases does not definitively describe a single service and might be a configuration problem
 // serviceAliasesResult_APIErrorHappened - means that 1 of N aliases got a 4xx/5xx and thereforce we cannot say 100% that the services doesn't exist
-// serviceAliasesResult_FoundServiceNoAlias - means that a service was found but that service has no alias (this should not be possible and can only happen from a bad code change.)
```

Comment:
This situation was added by me in an earlier PR and I verified it is not possible.